flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -159,6 +160,12 @@ private StreamingJobGraphGenerator(StreamGraph streamGraph, @Nullable JobID jobI
     private JobGraph createJobGraph() {
         preValidate();
 
+        if (streamGraph.shouldExecuteInBatchMode()) {
```

Comment:
I'm a bit worried that we're making decisions based on this flag in 2 different places, that must also be kept in sync.